package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,587 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""H5MD trajectories --- :mod:`MDAnalysis.coordinates.H5MD`
+========================================================
+
+The `H5MD`_ trajectory file format is based upon the general, high performance
+`HDF5`_ file format.
+HDF5 files are self documenting and can be accessed with the `h5py`_ library.
+The HDF5 library (and `H5PY`_) must be installed; otherwise, H5MD files
+cannot be read by MDAnalysis. If `H5PY`_ is not installed, a ``RuntimeError``
+is raised.
+
+HDF5 can make use of parallel file system features through the MPI-IO
+interface of the HDF5 library to improve parallel reads and writes.
+
+
+The `H5MD`_ file format is based upon `HDF5`_, which makes use of parallel
+file system features through the MPI-IO interface of the HDF5 library.
+The reader currently uses the `H5PY`_ library to access data from an H5MD file.
```

Comment:
Add an Examples section (might need to call it "Using H5MD" or similar because "Example" might be mangled by the numpy doc sphinx parser) that shows
1. opening a h5md file
2. opening a h5md in parallel (add the mpi4py commands for communicator etc)
3. using an existing `h5py.File`

Reply:
Regarding point 3, I tried writing a test to test if the reader could take an open `h5py.File` object, but the reader is failing because it's trying to open the file with h5py, which means its skipping the `isinstance(self.filename, h5py.File)` line. I'm not sure why it would skip this line, because even when I test the open file with `isintance` it returns `True`.

Reply:
for some reason, universe is reading the file object as a string

Reply:
I don't think it should: the following should work
```python
u = mda.Universe(h5py.File("sim.h5md"))
```
If it doesn't then ask on the developer list, please. This is part of the "converter" infrastructure https://docs.mdanalysis.org/2.0.0-dev0/documentation_pages/converters.html and _should work_. If not, then it might require a closer look or an issue to be raised. Ultimately, the `format_hint` should take care of everything.

Reply:
When I try to pass the argument directly like that, it doesn't work.

Following an example in https://www.mdanalysis.org/docs/documentation_pages/lib/util.html#containers-and-lists, I did
```
stream = h5py.File("trajectory.h5md", 'r')
u = mda.Universe("topology.tpr", NamedStream(stream, stream.filename))
```
and I was able to load the universe and access all of the timestep attributes. Is this a valid way of opening a stream?